server/service/src/document/mod.rs

```diff
@@ -1,28 +1,20 @@
-use repository::RepositoryError;
+use repository::{Document, RepositoryError};
 
 use crate::service_provider::{ServiceContext, ServiceProvider};
 
-use self::raw_document::RawDocument;
-
 pub mod document_registry;
 pub mod document_service;
 pub mod form_schema_service;
 pub mod raw_document;
 
-/// Checks that there is no document with a later timestamp
+/// Checks if the doc is the latest in the DB
 pub(crate) fn is_latest_doc(
     ctx: &ServiceContext,
     service_provider: &ServiceProvider,
-    doc: &RawDocument,
+    doc: &Document,
 ) -> Result<bool, RepositoryError> {
     let latest_existing = service_provider
         .document_service
         .document(ctx, &doc.name, None)?;
-    if let Some(lastest_existing) = latest_existing {
-        if lastest_existing.timestamp > doc.timestamp {
-            // newer doc already exist
-            return Ok(false);
-        }
-    }
-    Ok(true)
+    Ok(latest_existing.map(|e| e.id == doc.id).unwrap_or(false))
```

Comment:
Just looking at the code and method description, can't quite see how latest check it's done, it looks like `check if document id is unique for document name`

Reply:
Yes thanks, this is buggy, it has already been updated in a later PR where I compare the timestamp...

Reply:
to answer you comment better, it was working like this since every document version has a new id... However, there was the foot gun that this was assuming the document was already written which wasn't true in all cases.